<documentation title="Switch Declaration">
    <standard>
        <![CDATA[
        This standard covers all switch declarations.
        ]]>
    </standard>
    <standard>
        <![CDATA[
        <em>Case</em> statements are indented with a single indent (tab) inside the <em>switch</em> statement.
        ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Case statement is indent with one tab">
        <![CDATA[
switch ($something) {
<em>[tab]</em>case '1':
...
}
        ]]>
        </code>
        <code title="Invalid: Case statement is indent with one space. Case statement is not indent at all or with two tabs.">
        <![CDATA[
switch ($something) {
<em>[space]</em>case '1':
...
}

switch ($something) {
<em></em>case '1':
...
}

switch ($something) {
<em>[tab][tab]</em>case '1':
...
}
        ]]>
        </code>
    </code_comparison>
    <standard>
        <![CDATA[
        The code inside the <em>case</em> statements is further indented with a single (tab) indent.
        ]]>
    </standard>
    <code_comparison>
        <code title="Valid: The code is indent one more tab then case statement.">
        <![CDATA[
switch ($something) {
<em>[tab]</em>case '1':
<em>[tab][tab]</em>$case = '1';
...
}
        ]]>
        </code>
        <code title="Invalid: The code is indent with spaces or with only one tab.">
        <![CDATA[
switch ($something) {
<em>[tab]</em>case '1':
<em>[space][space]</em>$case = '1';
...
}

switch ($something) {
<em>[tab]</em>case '1':
<em>[tab]</em>$case = '1';
...
}

switch ($something) {
<em>[tab]</em>case '1':
<em>[tab][tab][tab]</em>$case = '1';
...
}
        ]]>
        </code>
    </code_comparison>
    <standard>
        <![CDATA[
        The <em>break</em> statement is aligned with the code.
        ]]>
    </standard>
    <code_comparison>
        <code title="Valid: The break statement is aligned to the code.">
        <![CDATA[
switch ($something) {
<em>[tab]</em>case '1':
<em>[tab][tab]</em>$case = '1';
<em>[tab][tab]</em>break;
...
}
        ]]>
        </code>
        <code title="Invalid: The break statement is not aligned with the code.">
        <![CDATA[
switch ($something) {
<em>[tab]</em>case '1':
<em>[tab][tab]</em>$case = '1';
<em>[tab]</em>break;
...
}

switch ($something) {
<em>[tab]</em>case '1':
<em>[tab][tab]</em>$case = '1';
<em>[tab][tab][tab]</em>break;
...
}
        ]]>
        </code>
    </code_comparison>
    <standard>
        <![CDATA[
        Only one <em>break</em> statement is allowed per <em>case</em>.
        ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Only one break statement per case">
        <![CDATA[
switch ($something) {
    case '1':
        $case = '1';
        <em>break;</em>
...
}
        ]]>
        </code>
        <code title="Invalid: Two case statements per case">
        <![CDATA[
switch ($something) {
    case '1':
        $case = '1';
        <em>break;
        break;</em>
...
}
        ]]>
        </code>
    </code_comparison>
    <standard>
        <![CDATA[
        The <em>default</em> statement must be the last in the <em>switch</em> and must not have a <em>break</em> statement.
        ]]>
    </standard>
    <code_comparison>
        <code title="Valid: The default statement is the last in the switch and have no break statement.">
        <![CDATA[
switch ($something) {
    case '1':
        $case = '1';
        break;
    default:
        $case = null;
}
        ]]>
        </code>
        <code title="Invalid: Default statement is not the last element and it contains a break.">
        <![CDATA[
switch ($something) {
    case '1':
        $case = '1';
        break;
    <em>default:
        $case = null;</em>
    case '2':
        $case = '2';
        break;
}

switch ($something) {
    case '1':
        $case = '1';
        break;
    default:
        $case = null;
        <em>break;</em>
}
        ]]>
        </code>
    </code_comparison>
    <standard>
        <![CDATA[
        If one <em>case</em> block has to pass control into another <em>case</em> block without having a <em>break</em>, there must be a comment about it in the code.
        ]]>
    </standard>
    <code_comparison>
        <code title="Valid: The fall-through is explained with a comment">
        <![CDATA[
switch ($something) {
    case '1':
        $case = '1';
        break;
    case '2':
        $case = '3';
        <em>// Fall through the next case on purpose</em>
    case '3':
        $case = '3';
        break;
    default:
        $case = null;
}
        ]]>
        </code>
        <code title="Invalid: No comment at the fall-through">
        <![CDATA[
switch ($something) {
    case '1':
        $case = '1';
        break;
    case '2':
        $case = '3';
        <em></em>
    case '3':
        $case = '3';
        break;
    default:
        $case = null;
}
        ]]>
        </code>
    </code_comparison>
</documentation>